File: src/services/UserService.ts
import { User } from '../models/User';
import { UserRepository } from '../repositories/UserRepository';

export class UserService {
  constructor(private userRepo: UserRepository) {}

  async getAllUsers(): Promise<User[]> {
    return this.userRepo.getAll();
  }

  async getUserById(userId: number): Promise<User | null> {
    return this.userRepo.getById(userId);
  }
}

File: CalendarApi/Services/UserService.cs
using CalendarApi.Models;
using CalendarApi.Repositories;

namespace CalendarApi.Services;

/// <summary>
/// Business logic for User operations.
/// Migrated from TypeScript UserService.ts
/// </summary>
public class UserService
{
    private readonly UserRepository _userRepo;

    public UserService(UserRepository userRepo)
    {
        _userRepo = userRepo;
    }

    public List<User> GetAllUsers()
    {
        return _userRepo.GetAll();
    }

    public User? GetUserById(int userId)
    {
        return _userRepo.GetById(userId);
    }
}
